cmd/notary-signer/config.go

```diff
@@ -82,20 +82,31 @@ func parseSignerConfig(configFilePath string, doBootstrap bool) (signer.Config,
 	}, nil
 }
 
-func getEnv(env string) string {
-	v := viper.New()
-	utils.SetupViper(v, envPrefix)
-	return v.GetString(strings.ToUpper(env))
-}
-
 func passphraseRetriever(keyName, alias string, createNew bool, attempts int) (passphrase string, giveup bool, err error) {
-	passphrase = getEnv(alias)
 
-	if passphrase == "" {
-		return "", false, errors.New("expected env variable to not be empty: " + alias)
+	// Construct a default passphrase store and passphrase protector.
+	// When more than one passphrase stores or passphrase protect methods are implemented,
+	// a decision will be made here, based on a config variable, as to which type to construct.
+	var passphraseStore signerpassp.Storage = signerpassp.NewDefaultPassphraseStore()
```

Comment:
@riyazdf yes as-is it does not deviate from the current behavior but is a step in the direction towards go plugins.